VideoLocker/src/main/java/org/edx/mobile/module/prefs/UserPrefs.java

```diff
@@ -51,20 +52,21 @@ public boolean isDownloadOverWifiOnly() {
      */
     public File getDownloadFolder() {
         ProfileModel profile = getProfile();
-        
-        File android = new File(Environment.getExternalStorageDirectory(), "Android");
-        File downloadsDir = new File(android, "data");
-        File packDir = new File(downloadsDir, context.getPackageName());
-        File edxDir = new File(packDir, profile.username);
-        edxDir.mkdirs();
+        File videosDir = new File(context.getExternalFilesDir(null).getParent(), AppConstants.Folders.VIDEOS);
```

Comment:
As far as writing to external storage, I think we should ONLY use `getExternalFilesDir`.

Like @1zaman mentioned, even if we could construct the path using another method, writing to that path would likely fail. So I would go with option (3) @1zaman mentioned further up: "Change the calling code to check for null and abort the download with an error message"

As for how to handle the upgrade, we also need to account for the case where we only did part of the migration, as the migration is non-transactional.

In either case, we must either:
1) Prevent the user from using the app / certain features until the migration completes (that doesn't sound great)
2) Allow the app / feature to continue to work even if the migration fails (support having the data in either location)

In either case, we can also decide when to attempt migration, either on app-open, user-input, and/or when we see `ACTION_MEDIA_MOUNTED`, etc.